src/Fixer/Comment/CommentToPhpdocFixer.php

```diff
@@ -52,8 +52,45 @@ public function isRisky(): bool
     /**
      * {@inheritdoc}
      *
-     * Must run before GeneralPhpdocAnnotationRemoveFixer, GeneralPhpdocTagRenameFixer, NoBlankLinesAfterPhpdocFixer, NoEmptyPhpdocFixer, NoSuperfluousPhpdocTagsFixer, PhpdocAddMissingParamAnnotationFixer, PhpdocAlignFixer, PhpdocAnnotationWithoutDotFixer, PhpdocInlineTagNormalizerFixer, PhpdocLineSpanFixer, PhpdocNoAccessFixer, PhpdocNoAliasTagFixer, PhpdocNoEmptyReturnFixer, PhpdocNoPackageFixer, PhpdocNoUselessInheritdocFixer, PhpdocOrderByValueFixer, PhpdocOrderFixer, PhpdocParamOrderFixer, PhpdocReturnSelfReferenceFixer, PhpdocSeparationFixer, PhpdocSingleLineVarSpacingFixer, PhpdocSummaryFixer, PhpdocTagCasingFixer, PhpdocTagTypeFixer, PhpdocToCommentFixer, PhpdocToParamTypeFixer, PhpdocToPropertyTypeFixer, PhpdocToReturnTypeFixer, PhpdocTrimConsecutiveBlankLineSeparationFixer, PhpdocTrimFixer, PhpdocTypesOrderFixer, PhpdocVarAnnotationCorrectOrderFixer, PhpdocVarWithoutNameFixer.
-     * Must run after AlignMultilineCommentFixer.
+     * Must run before:
+     *
+     * - GeneralPhpdocAnnotationRemoveFixer
+     * - GeneralPhpdocTagRenameFixer
+     * - NoBlankLinesAfterPhpdocFixer
+     * - NoEmptyPhpdocFixer
+     * - NoSuperfluousPhpdocTagsFixer
+     * - PhpdocAddMissingParamAnnotationFixer
+     * - PhpdocAlignFixer
+     * - PhpdocAnnotationWithoutDotFixer
+     * - PhpdocInlineTagNormalizerFixer
+     * - PhpdocLineSpanFixer
+     * - PhpdocNoAccessFixer
+     * - PhpdocNoAliasTagFixer
+     * - PhpdocNoEmptyReturnFixer
+     * - PhpdocNoPackageFixer
+     * - PhpdocNoUselessInheritdocFixer
+     * - PhpdocOrderByValueFixer
+     * - PhpdocOrderFixer
+     * - PhpdocParamOrderFixer
+     * - PhpdocReturnSelfReferenceFixer
+     * - PhpdocSeparationFixer
+     * - PhpdocSingleLineVarSpacingFixer
+     * - PhpdocSummaryFixer
+     * - PhpdocTagCasingFixer
+     * - PhpdocTagTypeFixer
+     * - PhpdocToCommentFixer
+     * - PhpdocToParamTypeFixer
+     * - PhpdocToPropertyTypeFixer
+     * - PhpdocToReturnTypeFixer
+     * - PhpdocTrimConsecutiveBlankLineSeparationFixer
+     * - PhpdocTrimFixer
+     * - PhpdocTypesOrderFixer
+     * - PhpdocVarAnnotationCorrectOrderFixer
+     * - PhpdocVarWithoutNameFixer
```

Comment:
and of course we have `getFixerWithFixedPosition`, that i think we do not even document here

Reply:
@keradus 

Maybe we should have an internal fixer that adds these DocBlocks?

Reply:
We have test that tests it - I'm OK with any approach, but I'm not sure if there is biggest value in automating applying the format. Oh wait, there is - because you want to change the format ;)

I think first things first - decide on format itself. We can come up with any format, the question is how useful this doc is.

Maybe we just dump the graph to single file (autogenerated), and we link that file here - random example that it does not need to be phpdoc+fixer.

myself, I do not rely on this comment at all - originally it was awesome to have it, but now it's that complex and long, that I actually skip the comment totally.

Reply:
Like the cleanup and it is a great improvement :+1: 

>I think first things first - decide on format itself.

Giving it some more thought, in order to figure out what format suits best we've to figure out what would suit the readers or audience of these docs best. Now I know I put the current format in place and all that, but overtime I started to wonder if listing these dependencies are really used and as such, maybe a `@see [ref. to the list of deps. in utest` would suffice

Reply:
Personally I would rather add `AbstraxtFixer::getPrecedingFixers(): array` and return collection of fixers that must be run before that one. This would allow creating dependency tree dynamically, test if there's dependency loop etc. Then we wouldn't need list of before/after in phpDoc and in the future we could even get rid of priorities, because fixers would be run in the order defined by their dependencies. Is it good odea to invest time in phpDoc format?

Reply:
there was some good discussion somewhere that it's nice to have it as code, but also it will be BC break for custom fixers that are relying on numbers (potentially we can have a transition period when both methods returning same results).

Reply:
Ovverall, agree to not overdo the docblock (eg with custom fixer), but rather keep it simple (eg externalised json file?) for short term (or even no changes) and auto-calculated priorities on long term

Reply:
> there was some good discussion somewhere that it's nice to have it as code, but also it will be BC break for custom fixers that are relying on numbers (potentially we can have a transition period when both methods returning same results).

I believe we could utilise current priorities for building dependency graph too, just by treating all fixers with lower priority as "dependencies" of fixers with higher priority. It would create huge list, but in the end it could be simplified to proper ordered list. I think the requirement for having integration test between fixers should be defined only by explicit dependency (using `AbstractFixer::getPrecedingFixers()`), so during implementation of such system we should migrate current dependency graph to the fixers' methods. This way we could keep support for custom fixers' priorities, but at the same time introduce new approach.

So, for transition period it would work like this:
- get "dependant" fixers by priority (all fixers with lower priority)
- get explicit dependencies defined in `AbstractFixer::getPrecedingFixers()`
- merge those lists (keep only unique values)
- calculate actual order of running, so each fixers is run after all dependant fixers

Then, in next major, we could drop support for priorities completely.

---

Current approach based on priorities is not good because:
- it's cumbersome to add new fixers between existing fixers when there's no unused priority between existing fixers (it requires changing priority to many other fixers just to move all of them up/down)
- term "priority" is not actually what it is. "Priority" for me relates to importance, but all fixers are equally important, they just need to be run in proper _order_ 😉.

---

In the future, fixers' order resolver could also introduce support for pre/post hooks, that would enable running some fixers multiple times for each file (once in calculated order, but also as a post-hook for other fixer that is applied later), which would completely resolve problems that currently happen, when user need to run Fixer more than once to get the final result. Maybe automatic order calculation would be efficient enough, but having it flexible and dynamic would allow some more tweaks for edge cases.

Reply:
I do not think this PR is good to define the approach for "getDependencies". I would love to find the previous discussion, but can't. maybe @kubawerlos or @SpacePossum remember the place?

> running some fixers multiple times

Reply:
> I do not think this PR is good to define the approach for "getDependencies"

Probably you're right, I just dump my thoughts about how it should be done properly, because I believe investing time in phpDoc formatting does not make much sense. I actually don't feel the need to have this phpDoc in first place, I never use this information _from there_. Explicit dependency graph should only require integration tests for fixers, not modifying phpDoc.

Anyway, format proposed by @localheinz is better in terms of git diff (just like multiline arrays or signatures), but has worse DX when you don't collapse phpDoc by default. So I'm pretty +0 about it 🤷‍♂️.

> > running some fixers multiple times
> 
> 👎🏻

As I said, dynamic order resolution would probably fix that "priority" problems anyway, so it was only example of how that resolver would allow us to hook into the final fixers' order. I understand that running fixer(s) multiple times would make the process longer 🙂.

Reply:
I've setup'ed the priorities in `FixerFactoryTest` in preparation for topological sorting, yet the fixed value ones and some exception cases (like the PHPDoc ones), circular dependencies and BC concerns made be stop there. 
It was discussed mostly off GH I think, there are some traces of the talks (for example https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6155#issuecomment-988579579), but might be better to start fresh?

Reply:
It seems like @kubawerlos [had exact same idea](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6155#issuecomment-988579579) as [me](#discussion_r1357254615), with slightly different naming and implementation details (new interface vs `AbstractFixer`). So it seems like a good direction 🙂.